consensus/ibft/consensus_backend.go

```
@@ -76,52 +71,48 @@ func (i *backendIBFT) InsertBlock(
 		return
 	}
 
-	if err := i.runHook(InsertBlockHook, header.Number, header.Number); err != nil {
-		i.logger.Error("cannot run hook", "name", string(InsertBlockHook), "err", err)
-
-		return
-	}
-
 	i.updateMetrics(newBlock)
 
 	i.logger.Info(
 		"block committed",
 		"number", newBlock.Number(),
 		"hash", newBlock.Hash(),
-		"validators", len(i.activeValidatorSet),
+		"validators", i.currentValidators.Len(),
 		"committed", len(committedSeals),
 	)
 
+	if err := i.currentHooks.PostInsertBlock(newBlock); err != nil {
+		return
+	}
+
 	// after the block has been written we reset the txpool so that
 	// the old transactions are removed
 	i.txpool.ResetWithHeaders(newBlock.Header)
 }
 
 func (i *backendIBFT) ID() []byte {
-	return i.validatorKeyAddr.Bytes()
+	return i.currentSigner.Address().Bytes()
 }
 
 func (i *backendIBFT) MaximumFaultyNodes() uint64 {
-	return uint64(i.activeValidatorSet.MaxFaultyNodes())
+	return uint64(CalcMaxFaultyNodes(i.currentValidators))
 }
 
 func (i *backendIBFT) Quorum(blockNumber uint64) uint64 {
-	var (
-		validators = i.activeValidatorSet
-		quorumFn   = i.quorumSize(blockNumber)
-	)
+	validators, _ := i.forkManager.GetValidators(blockNumber)
```

Why do you ignore the error here?

---

Fixed at 8cbf4e4923258170914ac9c445a9d7e17a5e007d

---

This is very dangerous. I don't think `0` is the correct option for Quorum as it allows blocks to be added without supervision.

cc @dbrajovic

---

@zivkovicmilos Then do you want me to return MaximumNumber? I think it's strange. Please consider to add error handling in go-ibft

---

> Please consider to add error handling in go-ibft
I'm sorry, but this doesn't make any sense.

The `Backend` interface is responsible for providing this value, and `0` is a dangerous value to provide if it errors out. Consider changing this to `MaxUint` or something similar

---

Fixed at 540b2e3f6dd01ecd5003b38cfcdebefbbb562b8e